include/libtorrent/bencode.hpp

```diff
@@ -328,7 +328,7 @@ namespace libtorrent
 			// ----------------------------------------------
 			// string
 			default:
-				if (is_digit(std::uint8_t(*in)))
+				if (is_digit(char(*in)))
```

Comment:
I see. would you mind adding a ``static_assert`` that ``sizeof(*in) == 1`` too then?